File: src/main/java/dev/slne/spawn/trader/entity/EntityInteractListener.java
package dev.slne.spawn.trader.entity;

import dev.slne.spawn.trader.SpawnTrader;
import dev.slne.spawn.trader.gui.SpawnTraderGUI;
import lol.pyr.znpcsplus.api.event.NpcInteractEvent;
import lol.pyr.znpcsplus.api.npc.NpcEntry;

import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.scheduler.BukkitRunnable;

public class EntityInteractListener implements Listener {

  @EventHandler
  public void onNpcInteract(NpcInteractEvent event) {
    NpcEntry entry = event.getEntry();
    Player player = event.getPlayer();

    if(entry.getId().equalsIgnoreCase("spawn-trader")) {
      new BukkitRunnable() {
        @Override
        public void run() {
          new SpawnTraderGUI(player).show(player);
        }
      }.runTaskLater(SpawnTrader.instance(), 0L);
    }
  }
}

File: src/main/java/dev/slne/spawn/trader/entity/EntityInteractListener.kt
package dev.slne.spawn.trader.entity

import com.github.shynixn.mccoroutine.folia.entityDispatcher
import com.github.shynixn.mccoroutine.folia.launch
import dev.slne.spawn.trader.entity.impl.TraderNPC
import dev.slne.spawn.trader.gui.SpawnTraderGUI
import dev.slne.spawn.trader.plugin
import dev.slne.surf.npc.api.event.NpcInteractEvent
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener

class EntityInteractListener : Listener {
    @EventHandler
    fun onNpcInteract(event: NpcInteractEvent) {
        val npc = event.npc

        if (npc == TraderNPC.npc) {
            plugin.launch(plugin.entityDispatcher(event.player)) {
                SpawnTraderGUI(event.player).show(event.player)
            }
        }
    }
}
